### PR TITLE
Report ErrorBoundary from Dashboard

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Adding a method for the dashbaord to report an error boundary via a new `reportErrorBoundary` rpc method. Internaly it:

* does a `log.info()` call w/ some metadata about the error (`url`, `message` and `userId`)
* increments a new prometheus counter, `gitpod_dashboard_error_boundary_total`, which has a `url` label.

## How to test

I was able to manually test this by forwarding the metrics server port to my workspace, and curl'ing the `/metrics` endpoint (hat-tip @geropl for the pointer).
```
kubectl port-forward deployment/server 9500 &
```
```
curl localhost:9500/metrics
```
After manually "breaking" it a few times by injecting some code that throws errors in a component render, was able to see the counters increment, and verify the `url` label is set as expected.
```
gitpod_dashboard_error_boundary_total{url="https://bmh-report12cfdef066.preview.gitpod-dev.com/workspaces"} 1
gitpod_dashboard_error_boundary_total{url="https://bmh-report12cfdef066.preview.gitpod-dev.com/workspaces?foo=bar"} 2
```

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@selfcontained: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -3568,6 +3568,11 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return [];
     }
 
+    async reportErrorBoundary(ctx: TraceContextWithSpan, url: string, message: string): Promise<void> {
+        log.info("dashboard error boundary", { message, url, userId: this.user?.id });
```
Comment (selfcontained) [inline components/server/src/workspace/gitpod-server-impl.ts:3572]: Wasn't sure if this should be a `log.error()` or `log.info()` so played it safe. Happy to adjust though.

Comment (easyCZ) [reply components/server/src/workspace/gitpod-server-impl.ts:3604]: Would be good to limit the size of the message and/or error as these contribute directly to our logging cost

File: components/server/src/prometheus-metrics.ts [hunk 2]
```diff
@@ -225,3 +225,15 @@ export function observespicedbClientLatency(
         durationInSeconds,
     );
 }
+
+export const dashboardErrorBoundary = new prometheusClient.Counter({
+    name: "gitpod_dashboard_error_boundary_total",
+    help: "Total number of errors caught by an error boundary in the dashboard",
+    labelNames: ["url"],
```
[the remark this answers is not in the document]
Comment (easyCZ) [reply components/server/src/prometheus-metrics.ts:232]: Alternatively, remove the URL entirely. You can push it into logs, but keep the metric without it.

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 3]
```diff
@@ -3600,6 +3600,11 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
...
```
Comment (easyCZ) [inline components/server/src/workspace/gitpod-server-impl.ts:3604]: I'd do Warn here. We can verify if it's a legit error, or abuse of the endpoint but assuming most invocations are legitimate, warn seems appropriate.

File: components/server/src/prometheus-metrics.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (selfcontained) [reply components/server/src/prometheus-metrics.ts:232]: Gotcha, that makes sense.. I'll drop it from the metric and we can rely on logs for now.

**Review by easyCZ — APPROVED** [review]

**Pull request merged**